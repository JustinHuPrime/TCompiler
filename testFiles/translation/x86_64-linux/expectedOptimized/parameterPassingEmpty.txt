testFiles/translation/x86_64-linux/input/parameterPassingEmpty.tc:
TEXT(_T3foo3bar,
  BLOCK(5,
    JUMP(LOCAL(3)),
  ),
  BLOCK(3,
    MOVE(TEMP(temp8, 8, 8, GP), GLOBAL(_T3foo3bar)),
    JUMP(LOCAL(7)),
  ),
  BLOCK(7,
    NOP(),
    CALL(TEMP(temp8, 8, 8, GP)),
    RETURN(),
  ),
)
