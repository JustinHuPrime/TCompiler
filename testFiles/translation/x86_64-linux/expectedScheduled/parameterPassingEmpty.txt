testFiles/translation/x86_64-linux/input/parameterPassingEmpty.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(0,
    LABEL(LOCAL(5)),
    LABEL(LOCAL(3)),
    MOVE(TEMP(temp8, 8, 8, GP), GLOBAL(_T3foo3bar)),
    LABEL(LOCAL(7)),
    NOP(),
    CALL(TEMP(temp8, 8, 8, GP)),
    RETURN(),
  ),
)
