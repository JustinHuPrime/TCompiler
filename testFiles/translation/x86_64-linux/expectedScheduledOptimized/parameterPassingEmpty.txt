testFiles/translation/x86_64-linux/input/parameterPassingEmpty.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(0,
    MOVE(TEMP(temp8, 8, 8, GP), CONSTANT(8, GLOBAL(_T3foo3bar))),
    NOP(),
    CALL(TEMP(temp8, 8, 8, GP)),
    RETURN(),
  ),
)
