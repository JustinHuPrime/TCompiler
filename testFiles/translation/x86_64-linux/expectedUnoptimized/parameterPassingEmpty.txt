testFiles/translation/x86_64-linux/input/parameterPassingEmpty.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(5,
    JUMP(CONSTANT(8, LOCAL(3))),
  ),
  BLOCK(3,
    MOVE(TEMP(temp8, 8, 8, GP), CONSTANT(8, GLOBAL(_T3foo3bar))),
    JUMP(CONSTANT(8, LOCAL(7))),
  ),
  BLOCK(7,
    NOP(),
    CALL(TEMP(temp8, 8, 8, GP)),
    JUMP(CONSTANT(8, LOCAL(4))),
  ),
  BLOCK(4,
    RETURN(),
  ),
)
