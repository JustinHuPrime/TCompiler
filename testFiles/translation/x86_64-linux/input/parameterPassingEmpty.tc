module foo;

void bar() {
  bar();
}