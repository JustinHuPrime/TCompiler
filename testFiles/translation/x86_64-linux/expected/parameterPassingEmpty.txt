testFiles/translation/x86_64-linux/input/parameterPassingEmpty.tc:
TEXT(_T3foo3bar,
  BLOCK(5,
    JUMP(LABEL(.L3)),
  ),
  BLOCK(3,
    JUMP(LABEL(.L4)),
  ),
  BLOCK(4,
    RETURN(),
  ),
)
